sources/packages/backend/apps/api/src/services/application-bulk-withdrawal/application-bulk-withdrawal-import-text.service.ts

```diff
@@ -115,6 +133,102 @@ export class ApplicationWithdrawalImportTextService {
+      applicationNumbers,
+      institutionId,
+      originator,
+    );
+    applicationValidationData.forEach((record) => {
+      applicationDataMap[record.applicationNumber] = {
+        applicationStatus: record.applicationStatus,
+        isArchived: record.isArchived,
+        sin: record.student.sinValidation.sin,
+        locationId: record.location.id,
+        locationCode: record.location.institutionCode,
+        hasPreviouslyBeenWithdrawn: !!record.studentScholasticStandings.length,
+      };
+    });
+    return textValidations.map((textValidation) => {
+      const validationModel = {} as ApplicationBulkWithdrawalValidationModel;
+      const applicationData =
+        applicationDataMap[textValidation.textModel.applicationNumber];
+      validationModel.sin = textValidation.textModel.sin;
+      validationModel.applicationNumber =
+        textValidation.textModel.applicationNumber;
+      validationModel.withdrawalDate = textValidation.textModel.withdrawalDate;
+      validationModel.applicationFound = false;
+      if (applicationData) {
+        validationModel.applicationFound = true;
+        validationModel.studentSINMatch =
+          textValidation.textModel.sin === applicationData.sin;
+        validationModel.isArchived = applicationData.isArchived;
+        validationModel.applicationStatus = applicationData.applicationStatus;
+        validationModel.hasPreviouslyBeenWithdrawn =
+          applicationData.hasPreviouslyBeenWithdrawn;
+      }
+      return validationModel;
+    });
+  }
+
+  /**
+   * Get the application validation details for all the applications.
+   * @param applicationNumbers application numbers for which the application details need to be retrieved.
+   * @param institutionId institution id.
+   * @param originator institution that sent the file.
+   * @returns applications containing the required information.
+   */
+  private async getApplicationValidationData(
+    applicationNumbers: string[],
+    institutionId: number,
+    originator: string,
+  ): Promise<Application[]> {
+    return this.applicationRepo.find({
+      select: {
+        id: true,
+        applicationNumber: true,
+        isArchived: true,
+        applicationStatus: true,
+        location: { id: true, institutionCode: true },
+        student: { id: true, sinValidation: { sin: true } },
+        studentScholasticStandings: {
+          id: true,
+        },
+      },
+      relations: {
+        location: true,
+        student: { sinValidation: true },
+        studentScholasticStandings: true,
+      },
+      where: {
+        applicationNumber: In(applicationNumbers),
+        location: {
+          institutionCode: originator,
+          institution: { id: institutionId },
+        },
+        studentScholasticStandings: {
+          changeType:
+            StudentScholasticStandingChangeType.StudentWithdrewFromProgram,
+        },
+      },
```

Comment:
I am having a hard time understanding the logic,  this method is to retrieve the info of the application that are requested for scholastic standing right? then why this logic? so, the ideal scenario is the application won't have any previous scholastic standing, so with this `where` condition, those application info won't be retrieving right? Let me know if I am missing anything

Reply:
Yes. Thanks @ann-aot . Corrected it.